src/Marker.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  Marker.swift
+//  ios-sdk
+//
+//  Created by Sarah Lensing on 3/22/17.
+//  Copyright © 2017 Mapzen. All rights reserved.
+//
+
+import Foundation
+import TangramMap
+
+public enum MarkerType : Int {
+  case currentLocation, searchPin, routeLine
+}
+
+@objc(MZMarker)
+public class Marker: NSObject {
+
+  private static let kPointStyle = "points"
+  private static let kLineStyle = "lines"
+  private static let kPolygonStyle = "polygons"
+  private static let kDefaultBackgroundColor = UIColor.white
+  private static let kDefaultInteractive = true
+  private static let kDefaultSize = CGSize.zero
+
+  public let tgMarker: TGMarker
+  private var styleType = kPointStyle
+  private var markerType: MarkerType?
+  // all marker types have an associated styling path
+  private static let typeToStylingPath = [MarkerType.currentLocation : "layers.mz_current_location_gem.draw.ux-location-gem-overlay",
+                                   MarkerType.searchPin : "layers.mz_search_result.draw.ux-icons-overlay",
+                                   MarkerType.routeLine : "layers.mz_route_line.draw.ux-route-line-overlay"]
+  //currently only search results have an inactive state
+  private static let typeToInactiveStylingPath = [MarkerType.searchPin : "layers.mz_search_result.inactive.draw.ux-icons-overlay"]
+
+  open var point: TGGeoPoint? {
+    set {
+      guard let p = newValue else { return }
```

Comment:
Yeah because it's an optional.

I probably attached my comment to the wrong thing here - but the basic question I have is for all of these (like the poly lines and polygons) is if we want to remove the poly line what's the correct way to go about it? Setting `nil` seems to be the correct way in iOS land, but would we rather nil out the entire marker if that's the case?